### ISSUE TITLE
block-scoped-var gives unnecessary warnings with multiple for loops

### ISSUE THREAD

**Issue by kasperpeulen** [eslint/eslint#7600]
**Tell us about your environment**

* **ESLint Version:** 3.9.1
* **Node Version:** v6.9.1
* **npm Version:** 3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```json
{
    "parserOptions": {
        "ecmaVersion": 5
    },
    "rules": {
        "block-scoped-var": "error"
    }
}
```

The following gives a warning:
```javascript
var list = [1, 2, 3];

for (var i = 0; i <list.length; i++) {
    console.log(i);
}
for (var i = 0; i <list.length; i++) {
    console.log(i);
}
```

I don't think this should be a warning. I think it should be warning if I use something like this:

```javascript
var list = [1, 2, 3];

for (var i = 0; i <list.length; i++) {
    console.log(i);
}
for (var i = 0; i <list.length; i++) {
    console.log(i);
}
console.log(i); // out of scope
```

Also the following code seems okay to me:

```javascript
if (1+1 === 2) {
  var message = "the universe is still sane";
  console.log(message);
} else {
  var message = "ooh noes 😱";
  console.log(message);
}
```

I think it would be problematic if I did this:

```javascript
if (1+1 === 2) {
  var message = "the universe is still sane";
} else {
  var message = "ooh noes 😱";
}
console.log(message); // out of scope
```

My idea about this is that I expect this rule to fail if `let` would fail. So we can slowly move our codebase to ES6. Or do I misinterpret this rule? If so, I think the docs could be clearer about this. 

Thanks for your time, and the excellent project btw.

**Comment by mysticatea** [comment]
Thank you for this issue.

However, this is intentional behavior.

> The block-scoped-var rule generates warnings when variables are used outside of the block in which they were defined.
> http://eslint.org/docs/rules/block-scoped-var

The rule warns references from outside of the block that the variable is defined.
The rule does not emulate complete block-scoping.

If you want block-scoping emulation, maybe you can use [mysticatea/block-scoped-var](https://github.com/mysticatea/eslint-plugin#eslint-plugin-mysticatea) rule of my plugin.
